### PR TITLE
ref(lint): Fix Typos

### PR DESCRIPTION
**Pull request by mgaeta**
Collected typo fixes from multiple PRs. Potentially dangerous changes are marked in a special commit.

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/eventWaiter.tsx [hunk 1]
```diff
@@ -109,14 +109,14 @@ class EventWaiter extends React.Component<Props, State> {
...
-        key: 'first_event_recieved',
+        key: 'first_event_received',
         organization,
         project,
       });
     } else {
       firstIssue = firstEvent;
       recordAnalyticsFirstEvent({
-        key: 'first_transaction_recieved',
+        key: 'first_transaction_received',
```
Comment (dashed) [inline src/sentry/static/sentry/app/utils/eventWaiter.tsx:119]: This cannot be changed until this is changed https://github.com/getsentry/reload/blob/67d393e63089a6044c499e3795fae8d169425e5a/reload_app/events.py#L667

File: src/sentry/static/sentry/app/utils/eventWaiter.tsx [hunk 2]
```diff
@@ -109,14 +109,14 @@ class EventWaiter extends React.Component<Props, State> {
       firstIssue = issues.find((issue: Group) => issue.firstSeen === firstEvent) || true;
 
       recordAnalyticsFirstEvent({
-        key: 'first_event_recieved',
+        key: 'first_event_received',
```
Comment (dashed) [inline src/sentry/static/sentry/app/utils/eventWaiter.tsx:112]: Cannot be changed until this is changed https://github.com/getsentry/reload/blob/67d393e63089a6044c499e3795fae8d169425e5a/reload_app/events.py#L663

**Comment by ceorourke** [comment]
@mgaeta you're doing the lord's work 👼

File: src/sentry/incidents/tasks.py [hunk 3]
```diff
@@ -110,7 +110,7 @@ def handle_snuba_query_update(subscription_update, subscription):
     """
     from sentry.incidents.subscription_processor import SubscriptionProcessor
 
-    with metrics.timer("incidents.subscription_procesor.process_update"):
+    with metrics.timer("incidents.subscription_processor.process_update"):
```
Comment (dashed) [inline src/sentry/incidents/tasks.py:113]: This might be dangerous to change.

File: src/sentry/static/sentry/app/utils/eventWaiter.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/eventWaiter.tsx:119]: Also I just wouldn't change it since there's no way to join that data

**Comment by mgaeta** [comment]
I reverted the changes with bad side effects and left an IDE comment `// noinspection SpellCheckingInspection` so that they won't be picked up again. Should I also put in something more human-readable?

**Review by dashed — APPROVED** [review]

**Comment by dashed** [comment]
@mgaeta looks good to me 👍

**Pull request merged**